### PR TITLE
Fix decomp behaviour in training IR

### PR DESCRIPTION
**Pull request by tugsbayasgalan**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #132722
* #132703
* #132033

Title

Differential Revision: [D60572995](https://our.internmc.facebook.com/intern/diff/D60572995/)

### CODE REVIEW

File: torch/_subclasses/fake_impls.py [hunk 1]
```diff
@@ -362,6 +362,7 @@ def repeat_interleave_tensor(fake_mode, func, repeats, output_size=None):
     return repeats.new_empty(output_size)
 
 
+@register_op_impl(torch.ops.aten.item.default)
```
Comment (tugsbayasgalan) [inline torch/_subclasses/fake_impls.py:365]: @ezyang Is this ok? In export, we give option for user to preserve CIA ops in the inference IR. `aten.item` is one such CIA op. As a result, we need a fake rule for it while tracing.

**Comment by ezyang** [comment]
I'm endorsing the item change, no comments on the rest

File: torch/export/exported_program.py [hunk 2]
```diff
@@ -350,6 +350,7 @@ def _decompose_and_get_gm_with_new_signature_constants(
 
         fake_args_unwrapped = pytree.tree_unflatten(fake_args, mod._in_spec)
         fake_mode = detect_fake_mode(fake_args)
+        fake_mode = contextlib.nullcontext() if fake_mode is None else fake_mode
```
Comment (ydwu4) [inline torch/export/exported_program.py:353]: Can we use_detect_fake_mode_from_gm? Compared with plain detect_fake_mode, it also detect fake modes based onmeta["val"] for intermediate results. Maybe fake_mode won't be None after using it.

**Review by ydwu4 — APPROVED** [review]
Overall looks good to me.

**Pull request closed without merging**